codex-rs/core/src/parse_command.rs

```diff
@@ -0,0 +1,2045 @@
+use crate::bash::try_parse_bash;
+use crate::bash::try_parse_word_only_commands_sequence;
+use serde::Deserialize;
+use serde::Serialize;
+use shlex::split as shlex_split;
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+pub enum ParsedCommand {
+    Read {
+        cmd: Vec<String>,
+        name: String,
+    },
+    Ls {
```

Comment:
Can you use a more declarative name? For example, at some point, we should support `dir` on Windows, which is not `ls` the command, but is functionally equivalent?